crowdsourcehinter/crowdsourcehinter.py

```diff
@@ -0,0 +1,352 @@
+    that specifically address their mistake. Additionally, the hints that this Xblock shows
+    are created by the students themselves. This doc string will probably be edited later.
+    """
+    # Database of hints. hints are stored as such: {"incorrect_answer": {"hint": rating}}. each key (incorrect answer)
+    # has a corresponding dictionary (in which hints are keys and the hints' ratings are the values).
+    #
+    # Example: {"computerr": {"You misspelled computer, remove the last r.": 5}}
+    hint_database = Dict(default={}, scope=Scope.user_state_summary)
+    # Database of initial hints, set by the course instructor. hint_database will receive the hints inputted
+    # in initial_hints. Initial hints have a default rating of 0.
+    #
+    # Example: {"Jeorge Washington": "You spelled his first name wrong."}
+    initial_hints = Dict(default={}, scope=Scope.content)
+    # This is a list of incorrect answer submissions made by the student. this list is mostly used for
+    # when the student starts rating hints, to find which incorrect answer's hint a student voted on.
+    #
+    # Example: ["personal computer", "PC", "computerr"]
+    incorrect_answers = List([], scope=Scope.user_state)
+    # A dictionary of generic_hints. default hints will be shown to students when there are no matches with the
+    # student's incorrect answer within the hint_database dictionary (i.e. no students have made hints for the
+    # particular incorrect answer)
+    #
+    # Example: ["Make sure to check your answer for simple mistakes like typos!"]
+    generic_hints = List(default=[], scope=Scope.content)
+    # List of which hints have been shown to the student
+    # this list is used to prevent the same hint from showing up to a student (if they submit the same incorrect answers
+    # multiple times)
+    #
+    # Example: ["You misspelled computer, remove the last r."]
+    used = List([], scope=Scope.user_state)
+    # This is a dictionary of hints that have been reported. the values represent the incorrect answer submission, and the
+    # keys are the hints the corresponding hints. hints with identical text for differing answers will all not show up for the
+    # student.
+    #
+    # Example: {"desk": "You're completely wrong, the answer is supposed to be computer."}
+    reported_hints = Dict(default={}, scope=Scope.user_state_summary)
+    # This String represents the xblock element for which the hinter is running. It is necessary to manually
+    # set this value in the XML file under the format "hinting_element": "i4x://edX/DemoX/problem/Text_Input" .
+    # Setting the element in the XML file is critical for the hinter to work.
+    #
+    # TODO: probably should change the name from Element (problem_element? hinting_element?). Trying to
+    # just change the name didn't seem to operate properly, check carefully what is changed
+    Element = String(default="", scope=Scope.content)
```

Comment:
Minor: I agree about changing the name, both to be more descriptive, and to lower-case the name.